### PR TITLE
Kernel plugin: Support lzma/xz-compressed initrd.

### PR DESCRIPTION
**Pull request by kyrofa**
This is another attempt at fixing LP: [#1569337](https://bugs.launchpad.net/snapcraft/+bug/1569337) which uses the magic library instead of shelling out to `file`. It also adds some tests.

### CODE REVIEW

File: snapcraft/plugins/kernel.py [hunk 1]
```diff
@@ -179,9 +180,33 @@ def _unpack_generic_initrd(self):
...
+
+            # Support gzip (can be gzip or x-gzip)
+            if 'gzip' in mime_type:
+                decompressor = 'gzip'
+            # Support lzma/xz
+            elif any(x in mime_type for x in ('x-xz', 'x-lzma')):
+                decompressor = 'xz'
+            else:
+                raise RuntimeError(
+                    'initrd file type is unsupported: {!r}'.format(mime_type))
```
Comment (elopio) [inline snapcraft/plugins/kernel.py]: great!

File: snapcraft/plugins/kernel.py [hunk 2]
```diff
@@ -179,9 +180,33 @@ def _unpack_generic_initrd(self):
...
+            mime_type = mime_detector.file(tmp_initrd_path)
+            if not mime_type:
+                raise RuntimeError(
+                    'Unable to determine mime type for {!r}: {}'.format(
+                        tmp_initrd_path, os.strerror(mime_detector.errno())))
+            logger.debug('initrd mime_type: {} {}'.format(
+                tmp_initrd_path, mime_type))
+
+            # Support gzip (can be gzip or x-gzip)
+            if 'gzip' in mime_type:
```
Comment (elopio) [inline snapcraft/plugins/kernel.py]: I think this check should be explicit for gzip or x-gzip.
There could be a case where we find foo-gzip, and we should fail with the unsupported exception.

File: snapcraft/plugins/kernel.py (hunk 1 above)
Comment (elopio) [inline snapcraft/plugins/kernel.py]: oh, almost great. You are missing a test for this case :)

**Comment by elopio** [comment]
:+1: with two buts.

File: snapcraft/plugins/kernel.py (hunk 1 above)
Comment (kyrofa) [inline snapcraft/plugins/kernel.py]: Gah, you're too fast. Just pushed it up.

File: snapcraft/plugins/kernel.py (hunk 2 above)
Comment (kyrofa) [inline snapcraft/plugins/kernel.py]: Alright, agreed.

Comment (kyrofa) [inline snapcraft/plugins/kernel.py]: Done.

**Comment by kyrofa** [comment]
retest this please

**Pull request merged**